### DISCUSSION TITLE
[DAI 2023-2024] Présentation des travaux pratiques du 23.01.2024

### DISCUSSION THREAD

**Discussion by ludelafo** [heig-vd-dai-course/heig-vd-dai-course#196] [Announcements]
Bonjour @heig-vd-dai-course/students,

J'ai pu définir l'ordre de passage pour les présentations de demain, mardi 23.01.2024.

Vous trouverez l'ordre de passage ici: <https://heig-vd-dai-course.github.io/heig-vd-dai-course/23-practical-work-4/#15>.

Cette fois-ci, vous avez droit à 10 minutes de présentation et il n'y aura pas de temps pour des questions.

Dans la seconde partie du cours (dès 16h35), nous ferons un apéro pour fêter la fin du semestre et faire un récapitulatif du cours. Nous prendrons le temps de revoir les évaluations GAPS (que vous pouvez trouver sur le dépôt Git à l'adresse suivante : <https://github.com/heig-vd-dai-course/heig-vd-dai-course/tree/main/26-semester-review-and-exam-preparation>), vous présenter les attentes et modalités de l'examen et finalement comment vous y préparer. Nous vous recommandons de venir. 🙂

Bonne fin d'après-midi et à demain !
Ludovic et Hadrien

**Comment by apirakas** [comment]
Bonsoir @ludelafo  et @hadrylouis,

J'ai 2 questions concernant les critères d'évaluation:
Va-t-on être penalisé si nous n'effectuons pas de git clone HTTPS sur notre serveur ?
Doit-on publier notre image docker où uniquement montrer comment le faire suffit ?

Et suite à votre message je pense que le dernier point "The answers to the questions are correct" ne fait plus de sens.

Bonne soirée

**Reply by ludelafo** [reply]
Bonsoir @apirakas,

> Va-t-on être penalisé si nous n'effectuons pas de git clone HTTPS sur notre serveur ?

Oui, cela est mentionné dans le critère 8. Nous nous attendons à ce que vous cloniez votre projet sur le serveur pour ensuite le lancer et l'exécuter.

> Doit-on publier notre image docker où uniquement montrer comment le faire suffit ?

Oui, nous nous attendons à ce que votre image soit publiée (comme pour le travail pratique 3) pour que n'importe qui puisse déployer et utiliser votre service.

> Et suite à votre message je pense que le dernier point "The answers to the questions are correct" ne fait plus de sens.

Vous avez raison. Malheureusement le temps ne permettra sans doute pas à ce qu'on ait le temps pour des questions. Partez du principe qu'il s'agit de points offerts du coup. 🙂 Nous verrons l'année prochaine s'il y a assez de temps et si l'on se rend compte que ce n'est simplement pas possible, nous le remplacerons par un autre critère.

Est-ce que j'ai répondu à toutes vos questions ?

**Reply by apirakas** [reply]
Bonsoir @ludelafo,

Merci pour les réponses, alors nous sommes obligé de mettre notre repo ainsi que notre docker en public ?

**Reply by ludelafo** [reply]
Bonsoir @apirakas,

Oui, dans le contexte de ce travail pratique oui.

Nous vous recommanderons d'ailleurs demain de mettre tous vos projets publics. Vous nous avez toujours présenté de très chouettes travaux pratiques et même si ceux-ci restent effectués dans un cadre scolaire (on sait qu'il y a toujours quelque chose à améliorer), cela vous fait une jolie carte de visite, que ce soit pour le monde professionnel ou privé pour montrer ce que vous avez accompli.

Si vous souhaitez vraiment garder votre repo privé, vous pouvez utiliser un Personal Access Token (PAT) avec les bons droits comme vu précédemment dans le cours pour vous authentifier en HTTPS sans devoir saisir vos identifiants personnels: <https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens>.
